rasa/core/policies/policy.py

```diff
@@ -111,11 +111,17 @@ def supported_stack_frames() -> List[StackFrameType]:
         """Returns the stack frames supported by the policy."""
         return []
 
-    def supports_current_stack_frame(self, tracker: DialogueStateTracker) -> bool:
+    def supports_current_stack_frame(
+        self, tracker: DialogueStateTracker, only_after_user_message: bool = True
+    ) -> bool:
+        """Check whether the policy is allowed to act."""
         flow_stack = FlowStack.from_tracker(tracker)
 
         if top_frame := flow_stack.top():
             return top_frame.frame_type in self.supported_stack_frames()
+        elif only_after_user_message and len(tracker.events) > 0:
+            last_event_type = tracker.events[-1].type_name
+            return last_event_type == UserUttered.type_name
```

Comment:
I am not sure if this check always works. there might be other (even though odd) events after the useruttered.

i think the better check is to see if there is an action after the latest useruttered. i also think this is already implemented somewhere.

Update: found it `tracker.has_action_after_latest_user_message()`